src/embed-code/components/EmbedCodeOverview.tsx

```diff
@@ -163,15 +127,46 @@ const EmbedCodeOverview: FC<EmbedCodeOverviewProps & UserProps> = ({ commonUser
 		}
 	}, [commonUser, setEmbedCodes, setLoadingInfo, tText, debouncedFilters]); // eslint-disable-line
 
-	const removeEmbedCode = async (id: EmbedCode['id']) => {
-		if (!commonUser?.profileId) {
-			return;
-		}
+	const reloadEmbedCodes = async () => {
+		await fetchEmbedCodes();
+		onUpdate && onUpdate();
+	};
 
+	const duplicateSelectedEmbedCode = async (selected: EmbedCode) => {
+		await duplicateEmbedCode({
+			title: selected.title,
+			contentType: selected.contentType,
+			contentId: selected.contentId,
+			descriptionType: selected.descriptionType,
+			description: selected.description,
+			start: selected.start,
+			end: selected.end,
+			externalWebsite: selected.externalWebsite,
+		} as EmbedCode);
+
+		ToastService.success(tHtml('Het fragment werd succesvol gedupliceerd.'));
+
+		setSelected(undefined);
+		await reloadEmbedCodes();
+	};
+
+	const changeEmbedCode = async (data: EmbedCode) => {
 		try {
-			// TODO EMBED: delete code by id
-			await fetchEmbedCodes();
+			await updateEmbedCode(data);
+			ToastService.success(tText('Fragment succesvol gewijzigd.'));
+
+			await reloadEmbedCodes();
 
+			setIsEmbedCodeModalOpen(false);
+			setSelected(undefined);
+		} catch (err) {
+			ToastService.danger(tText('Fragment wijzigen mislukt'));
```

Comment:
also log the error for developers:
```suggestion
			console.error(err);
			ToastService.danger(tText('Fragment wijzigen mislukt'));
```